### ISSUE TITLE
FAB button colouring logic is wrong

### ISSUE THREAD

**Issue by TheLastProject** [CatimaLoyalty/Android#969]
The contrast should be the same, the icon should just be more dark mode focused or not.

Just compare the edit button on:
| | |
| - | - |
| ![Light](https://github.com/CatimaLoyalty/Android/blob/da01730c73cee713c6196cd3b5271535fc0884a7/fastlane/metadata/android/en-US/images/phoneScreenshots/screenshot-04.png?raw=true) | ![Dark](https://github.com/CatimaLoyalty/Android/blob/da01730c73cee713c6196cd3b5271535fc0884a7/fastlane/metadata/android/en-US/images/phoneScreenshots/screenshot-08.png?raw=true) |

(Important note: this has been broken for very long, I only noticed it now. It was NOT broken by #937).

**Comment by Kethen** [comment]
the current logic for the edit button icon color selection is the following:
https://github.com/CatimaLoyalty/Android/blob/ebbcbf324f32ef70ed111c3d98fa657fc2ac7c1a/app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java#L618

```
        int darkenedColor = ColorUtils.blendARGB(backgroundHeaderColor, Color.BLACK, 0.1f);
        ...
        int complementaryColor = Utils.getComplementaryColor(darkenedColor);
        editButton.setBackgroundTintList(ColorStateList.valueOf(complementaryColor));
        Drawable editButtonIcon = editButton.getDrawable();
        editButtonIcon.mutate();
        int colorPrimary = MaterialColors.getColor(this, R.attr.colorPrimary, ContextCompat.getColor(this, R.color.md_theme_light_primary));
        int colorOnPrimary = MaterialColors.getColor(this, R.attr.colorOnPrimary, ContextCompat.getColor(this, R.color.md_theme_light_onPrimary));
        boolean darkMode = Utils.isDarkModeEnabled(this);
        if (Utils.needsDarkForeground(complementaryColor)) {
            editButtonIcon.setTint(darkMode ? colorOnPrimary : colorPrimary);
        } else {
            editButtonIcon.setTint(darkMode ? colorPrimary : colorOnPrimary);
        }
        editButton.setImageDrawable(editButtonIcon);
```

Do you prefer it to just set black/white instead of retrieving a bright/dark color from the current material you theme? On the other hand, the current design guideline for FAB+bottom app bar looks like this https://m3.material.io/components/bottom-app-bar/overview

**Comment by TheLastProject** [comment]
Unless I'm mistaking, it seems like the white or black normally used for the icon of the button isn't pure white or black, but a slightly different shade. I'm having trouble finding that shade. May just go for a pure black or white foreground though if I can't find it (the background colour is fine).
